### PR TITLE
feat: Add per-environment default configurations

### PR DESCRIPTION
**Pull request by nand4011**
Add a configuration object that contains settings for gRPC connections. It will eventually contain retry and middleware configs as well.

Add Laptop, InRegion, and LowLatency default configurations.

### CODE REVIEW

File: momento-sdk/src/main/java/momento/sdk/CacheClientBuilder.java [hunk 1]
```diff
@@ -3,28 +3,40 @@
...
 
   private final CredentialProvider credentialProvider;
+  private Configuration configuration;
   private final Duration itemDefaultTtl;
-  private Duration requestTimeout = null;
 
   CacheClientBuilder(
-      @Nonnull CredentialProvider credentialProvider, @Nonnull Duration itemDefaultTtl) {
+      @Nonnull CredentialProvider credentialProvider,
+      Configuration configuration,
```
Comment (rishtigupta) [inline momento-sdk/src/main/java/momento/sdk/CacheClientBuilder.java:19]: Should the configuration be @Nonnull too?

Comment (nand4011) [reply momento-sdk/src/main/java/momento/sdk/CacheClientBuilder.java:19]: Good catch.

File: momento-sdk/src/main/java/momento/sdk/config/Configurations.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * Provides the latest recommended configuration for a dev environment.
+     *
+     * <p>This configuration may change in future releases to take advantage of improvements we
+     * identify for default configurations.
+     *
+     * @return the latest Laptop configuration
+     */
+    public static Configuration Latest() {
+      final TransportStrategy transportStrategy =
+          new StaticTransportStrategy(new GrpcConfiguration(Duration.ofMillis(15000)));
```
Comment (rishtigupta) [inline momento-sdk/src/main/java/momento/sdk/config/Configurations.java:31]: Can we put the harcoded millisecond values as a global variable?
LATEST_LAPTOP_MILLIS, IN_REGION_MILLIS, LOW_LATENCY_MILLIS - something along these lines. Might help we we want to change these values in future?

Comment (rishtigupta) [reply momento-sdk/src/main/java/momento/sdk/config/Configurations.java:31]: Got it, thanks for clarifying!

**Review by rishtigupta — APPROVED** [review]

**Pull request merged**